songhyun/app.js

```diff
@@ -49,6 +50,50 @@ app.post("/join", async (req, res) => {
   res.status(201).json({ message: "user created" });
 });
 
+app.post("/login", async (req, res) => {
+  const { name, password } = req.body;
+
+  const hashedPassword = await myDataSource.query(
+    `SELECT 
+      users.password,
+      users.id
+    FROM users
+    WHERE users.name="${name}";
+    `
+  );
+  const dbPassword = hashedPassword[0].password;
+  const userId = hashedPassword[0].id;
+  const check = await bcrypt.compare(password, dbPassword);
+  if (!check) {
+    res.json({ message: "Invalid User" });
+  }
+
+  const send = { userId: userId };
+  const jwtToken = jwt.sign(send, process.env.secretKey);
+  res.json({ accessToken: jwtToken });
+});
```

Comment:
```suggestion
  const [user] = await myDataSource.query(
    `SELECT 
      users.password,
      users.id
    FROM users
    WHERE users.name="${name}";
    `
  );

  const check = await bcrypt.compare(password, user.password);
  if (!check) {
    res.json({ message: "Invalid User" });
  }

  const payload = { userId: user.id };
  const accessToken = jwt.sign(payload, process.env.secretKey);
  res.json({ accessToken });
});
```
불필요한 변수 선언을 줄이고 조금 더 간결하게 작성해 볼 수 있을 것 같아요.